Formula/lib/libnghttp3.rb

```diff
@@ -0,0 +1,37 @@
+class Libnghttp3 < Formula
+  desc "HTTP/3 library written in C"
+  homepage "https://nghttp2.org/nghttp3/"
+  url "https://github.com/ngtcp2/nghttp3/archive/refs/tags/v0.15.0.tar.gz"
+  sha256 "6a75f5563e58d99e6b98d442111ac677984011c66b8b4f923764712741399027"
+  license "MIT"
+  head "https://github.com/ngtcp2/nghttp3.git", branch: "main"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :test
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args, "-DENABLE_LIB_ONLY=1"
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    (testpath/"test.c").write <<~EOS
+      #include <nghttp3/nghttp3.h>
+      #include <stdio.h>
+
+      int main(void) {
+        nghttp3_qpack_decoder *decoder;
+        if (nghttp3_qpack_decoder_new(&decoder, 4096, 0, nghttp3_mem_default()) != 0) {
```

Comment:
Reviewer note: these parameters were chosen based on the recommendations from https://nghttp2.org/nghttp3/qpack-howto.html#using-qpack-decoder